source/slang/lexer.cpp

```diff
@@ -14,7 +14,7 @@ namespace Slang
 {
     static Token GetEndOfFileToken()
     {
-        return Token(TokenType::EndOfFile, "", SourceLoc());
+        return Token(TokenType::EndOfFile, UnownedStringSlice::fromLiteral(""), SourceLoc());
```

Comment:
Is there a reason to favor a `fromLiteral` operation here instead of just using `UnownedTerminatedStringSlice("")`?

Also: doesn't the default constructor for `UnownedStringSlice` produce an empty string anyway (just with both pointers being null)?

Reply:
On first point - because being string literal I can just get the length from the array length - 1. With UnownedTerminatedStringSlice it uses strlen. Now you could argue that the compiler will optimize that away on release, and it typically may. Moreover though I think (unless there is some special reason not to) using construction methods for a single type is better than creating new types to specially construct. Why? Because the methods and thus usage are grouped together (and easy to find), and all things being equal less types is better than more. So having UnownedStringSlice::makeTerminated(..) I think is probably superior. 

On default UnownedStringSlice, we could. We may want to distinguish between an 'empty' string and a 'never set' string or 'invalid' string. And nullptr could do that. We don't do that here, so it is arguably equivalent for our current usage.